cli/pack/archive.go

```diff
@@ -0,0 +1,154 @@
+package pack
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"github.com/apex/log"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// ArchivePacker is a structure that implements Packer interface
+// with specific archive packing behavior.
+type ArchivePacker struct {
+	opts *Opts
+}
+
+func NewArchivePacker(opts *Opts) *ArchivePacker {
+	packer := &ArchivePacker{
+		opts: opts,
+	}
+	return packer
+}
+
+// Pack of Archiver packs the bundle into tarball.
+func (packer *ArchivePacker) Pack() error {
+	path, err := pack(packer.opts)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(path)
+	log.Infof("The passed bundle packed into temporary directory: %s", path)
+
+	startPath, _ := os.Getwd()
+	defer func() {
+		_ = os.Chdir(startPath)
+	}()
+
+	err = os.Chdir(filepath.Dir(packer.opts.CmdCtx.Cli.ConfigPath))
+	if err != nil {
+		return err
+	}
+
+	var tarName, packageVersion string
+	if packer.opts.PackFlags.Version == "" {
+		packageVersion, err =
+			checkVersionFromGit(packer.opts.CliOpts.App.InstancesAvailable)
+		if err != nil {
+			packageVersion = "0.1.0.0"
+		}
+	} else {
+		packageVersion = packer.opts.PackFlags.Version
+	}
+
+	if packer.opts.PackFlags.Name != "" {
+		tarName = packer.opts.PackFlags.Name + "_" + packageVersion + ".tar.gz"
+	} else if packer.opts.PackFlags.FileName != "" {
+		tarName = packer.opts.PackFlags.FileName
+	} else {
+		absPath, _ := filepath.Abs(".")
```

Comment:
You made chdir to config path directory on line 41. What name will have the package? The name of the directory where config is located?

Reply:
Yes, considered as the name of bundle.